src/sinks/datadog/metrics/encoder.rs

```diff
@@ -106,6 +112,7 @@ impl Default for EncoderState {
             // conversion trait implementations that are also only none vs gzip.
             writer: get_compressor(),
             written: 0,
+            event_bytes_written: 0,
```

Comment:
I don't believe you actually need a different field here.

`written` already tracks the number of uncompressed bytes we've fed into the compressor, because the encoder separately tracks compressed and decompressed size, as Datadog has distinct limits for payloads both in their compressed and decompressed state.

Reply:
Yeah this was only needed when I was tracking the bytes without the header and footer.